src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/DatasourceExtension.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.jobscheduler;
+
+import org.opensearch.jobscheduler.spi.JobSchedulerExtension;
+import org.opensearch.jobscheduler.spi.ScheduledJobParser;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+
+/**
+ * Datasource job scheduler extension
+ */
+public class DatasourceExtension implements JobSchedulerExtension {
```

Comment:
what this extension is doing please add that in comment

Reply:
Added